File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int buy = prices[0];
        int profit = 0;
        if(n<=1) return 0;
        
        for(int i=1; i<n; i++){
            if(prices[i] <= buy){
                buy = prices[i];
                continue;
            }
            profit = Math.max(profit, prices[i] - buy);
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini = prices[0]
        profit = 0
        n = len(prices)
        for i in range(1,n):
            cost = prices[i] - mini
            profit = max(profit, cost)
            mini = min(mini,prices[i])
        return profit
        